pypit/arms.py

```diff
@@ -53,10 +66,32 @@ def ARMS(spectrograph, fitstbl, setup_dict):
     basenames = [None]*numsci  # For fluxing at the very end
 
     # Spectrometer class
-    spectro_class = spectro_utils.load_spec_class(spectrograph=spectrograph)
+    if spectrograph is None:
+        # Set spectrograph from FITS table instrument header
+        # keyword.
+        if par is not None and par['rdx']['spectrograph'] != fitstbl['instrume'][0]:
+            msgs.error('Specified spectrograph does not match instrument in the fits table!')
+        _spectrograph = load_spectrograph(spectrograph=fitstbl['instrume'][0])
+    elif isinstance(spectrograph, basestring):
+        _spectrograph = load_spectrograph(spectrograph=spectrograph)
+    elif isinstance(spectrograph, Spectrograph):
+        _spectrograph = spectrograph
+    else:
+        raise TypeError('Could not instantiate Spectrograph!')
+
+    # Instantiate the parameters
+    _par = _spectrograph.default_pypit_par() if par is None else par
+    if not isinstance(_par, pypitpar.PypitPar):
+        raise TypeError('Input parameters must be a PypitPar instance.')
+    required = [ 'rdx', 'calibrations', 'scienceframe', 'objects', 'extract', 'skysubtract',
```

Comment:
A minor comment... I like that the top-level arguments are spelled out, but I think we need to think about changing `rdx` to something a little more transparent.

Reply:
I figured out a way around the confusion in my previous implementation before that was confusing this with the python function reduce.  So I think we could change it back to reduce, if that's better.